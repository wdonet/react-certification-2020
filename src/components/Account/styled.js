import styled from 'styled-components';
import { AccountCircle } from '@styled-icons/material-outlined/AccountCircle';

const AccountIcon = styled(AccountCircle)`
  color: white;
  width: 50px;
  height: 50px;
  float: right;
  margin-left: 30px;
`;

const Dropdown = styled.ul`
  position: absolute;
  display: ${({ openDrop }) => (openDrop ? 'block' : 'none')};
  padding-inline-start: 0px;
  background-color: lightgray;
  padding: 8px;
  margin-top: 40px;
`;

const Button = styled.button`
  box-sizing: inherit;
  background-color: transparent;
  font-size: 19px;
  color: white;
  font-weight: bold;
  border: none;
`;

export { AccountIcon, Dropdown, Button };
