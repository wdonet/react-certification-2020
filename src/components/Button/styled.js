import styled from 'styled-components';

const StyledButton = styled.button`
  margin-right: 16px;
  color: inherit;
  margin-left: -12px;
  flex: 0 0 auto;
  padding: 12px;
  overflow: visible;
  font-size: 1.5rem;
  text-align: center;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 50%;
  border: 0;
  cursor: pointer;
  margin: 0;
  display: inline-flex;
  outline: 0;
  position: relative;
  align-items: center;
  user-select: none;
  vertical-align: middle;
  justify-content: center;
  text-decoration: none;
  background-color: transparent;
  -webkit-tap-highlight-color: transparent;
`;

export { StyledButton };
