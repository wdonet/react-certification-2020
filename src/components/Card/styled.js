import styled from "styled-components";

const Card = styled.div`
    width: 32%;
    float: left;
    background: #FFF;
    margin-right: 14px;
    min-height: 490px;
    margin-bottom: 14px;
    box-shadow: 0px 2px 1px -1px rgb(0 0 0 / 20%), 0px 1px 1px 0px rgb(0 0 0 / 14%), 0px 1px 3px 0px rgb(0 0 0 / 12%);
`;

const CardImg = styled.div`
    min-height: 180px;
    background-position: center top;
    background-size: auto 100%;
    background-image: url(${props => props.imgUrl});
    cursor: pointer;
`;

const CardContent = styled.div`
    display: grid;
    width: 100%;
    padding: 20px;
`;

const CardTitle = styled.h3`
    font-size: 1.25rem;
    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
    font-weight: 500;
    line-height: 1.6;
    letter-spacing: 0.0075em;
    cursor: pointer;
`;

const CardDescription = styled.p`
    font-size: 0.875rem;
    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
    font-weight: 400;
    line-height: 1.43;
    letter-spacing: 0.01071em;
    color: rgba(0, 0, 0, 0.54);
`;

const Styled = { Card, CardImg, CardContent, CardTitle, CardDescription };
export default Styled;