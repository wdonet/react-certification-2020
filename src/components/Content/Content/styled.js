import styled from 'styled-components';

const Content = styled.div`
  width: 100%;
  height: 100%;
`;

const SideBar = styled.div`
  float: left;
  width: 25%;
  height: 100%;
`;

const Main = styled.div`
  float: right;
  width: 75%;
  height: 100%;
  margin-top: 120px;
`;

const Styled = { Content, SideBar, Main };
export default Styled;
