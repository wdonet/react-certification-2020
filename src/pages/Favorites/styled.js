import styled from 'styled-components';

const StyledSection = styled.section`
  text-align: center;
`;

const Title = styled.h2`
  color: ${({ theme }) => theme.font.primary};
`;

export { StyledSection, Title };
