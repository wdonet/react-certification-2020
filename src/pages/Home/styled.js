import styled from "styled-components";

const Title = styled.h1`
    font-size: 1.8rem;
    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
    font-weight: 500;
    line-height: 1.9;
    letter-spacing: 0.0075em;
    color:#FFF;
    text-align: center;
`;

const HomeGrid = styled.div`
    display: inline-block;
    width: 100%;
    padding: 0 20px 10px;
`;

const Styled = { Title, HomeGrid };
export default Styled;