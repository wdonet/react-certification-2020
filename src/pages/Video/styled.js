import styled from 'styled-components';

const StyledSection = styled.section`
  padding: 30px;
  display: flex;
  justify-content: space-around;
`;

const RelatedVideosContainer = styled.div`
  padding: 15px;
  background-color: ${({ theme }) => theme.background.secondary};
  box-shadow: 0px 1px 10px 0px rgb(0 0 0 / 12%);
  display: flex;
  flex-direction: column;
  width: 400px;
  @media (max-width: 900px) {
    width: 350px;
  }
  @media (max-width: 700px) {
    width: 300px;
  }
`;

const VideoPlayerContainer = styled.div`
  width: 60%;
  background-color: ${({ theme }) => theme.background.secondary};
  box-shadow: 0px 1px 10px 0px rgb(0 0 0 / 12%);
  display: flex;
  flex-direction: column;
  padding: 20px;
`;

const RelatedVideoTitle = styled.p`
  margin: 10px 0 20px 5px;
  font-size: 1.4em;
  font-weight: 400;

  color: ${({ theme }) => theme.font.secondary};
`;

const TitleContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

const MainVideoTitle = styled.p`
  font-size: 1.5em;
  font-weight: 400;

  color: ${({ theme }) => theme.font.primary};
`;

const MainVideoDescription = styled.p`
  font-size: 0.8em;
  color: ${({ theme }) => theme.font.secondary};
`;

export {
  StyledSection,
  RelatedVideosContainer,
  VideoPlayerContainer,
  RelatedVideoTitle,
  MainVideoTitle,
  MainVideoDescription,
  TitleContainer,
};
