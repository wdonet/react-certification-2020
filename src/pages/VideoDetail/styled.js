import styled from 'styled-components';

const VideoDetail = styled.div`
  width: 100%;
  background-color: ${({ theme }) => theme.content};
`;

const Player = styled.div`
  width: 65%;
  height: 700px;
  float: left;
  position: relative;
  @media screen and (max-width: 768px) {
    width: 100%;
    height: 350px;
  }
`;

const List = styled.div`
  width: 35%;
  float: right;
  position: relative;
  @media screen and (max-width: 768px) {
    display: none;
  }
`;

const VideoInfo = styled.div`
  width: 100%;
`;

const VideoDescription = styled.div`
  width: 75%;
  height: 700px;
  float: left;
  position: relative;
  color: ${({ theme }) => theme.text};
`;

const AddFavorites = styled.div`
  width: 25%;
  float: right;
  position: relative;
  text-align: right;
  padding-top: 20px;
  color: ${({ theme }) => theme.text};
`;

export { Player, VideoDetail, List, VideoInfo, VideoDescription, AddFavorites };
