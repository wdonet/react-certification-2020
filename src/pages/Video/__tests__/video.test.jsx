import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';

import VideoPage from '../index';
import { StyledLink } from '../../../components/HeaderButton/styled';
import AuthProvider from '../../../providers/Auth';
import YoutubeDataProvider from '../../../providers/YoutubeData';
import GlobalProvider from '../../../providers/Global';

import { lightTheme } from '../../../theme';

describe('VideoPage', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <GlobalProvider>
            <AuthProvider>
              <YoutubeDataProvider>
                <ThemeProvider theme={lightTheme}>
                  <VideoPage />
                </ThemeProvider>
              </YoutubeDataProvider>
            </AuthProvider>
          </GlobalProvider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('adds videos to `favorites` correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <GlobalProvider>
          <AuthProvider authenticatedJest>
            <YoutubeDataProvider>
              <ThemeProvider theme={lightTheme}>
                <VideoPage />
              </ThemeProvider>
            </YoutubeDataProvider>
          </AuthProvider>
        </GlobalProvider>
      </BrowserRouter>
    );

    const Button = container.getElementsByClassName(StyledLink.styledComponentId)[0];

    fireEvent.click(Button);

    expect(Button).toBeTruthy();
  });
});
