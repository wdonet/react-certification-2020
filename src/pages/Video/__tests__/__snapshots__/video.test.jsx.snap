// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPage renders correctly 1`] = `
.c2 {
  width: 100%;
  height: 100%;
  min-height: 380px;
}

.c0 {
  padding: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c6 {
  padding: 15px;
  background-color: #FFFFFF;
  box-shadow: 0px 1px 10px 0px rgb(0 0 0 / 12%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 400px;
}

.c1 {
  width: 60%;
  background-color: #FFFFFF;
  box-shadow: 0px 1px 10px 0px rgb(0 0 0 / 12%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
}

.c7 {
  margin: 10px 0 20px 5px;
  font-size: 1.4em;
  font-weight: 400;
  color: #666666;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  font-size: 1.5em;
  font-weight: 400;
  color: #333333;
}

.c5 {
  font-size: 0.8em;
  color: #666666;
}

@media (max-width:900px) {
  .c6 {
    width: 350px;
  }
}

@media (max-width:700px) {
  .c6 {
    width: 300px;
  }
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      id="player"
    />
    <div
      className="c3"
    >
      <p
        className="c4"
      >
        
      </p>
    </div>
    <p
      className="c5"
    >
      
    </p>
  </div>
  <div
    className="c6"
  >
    <p
      className="c7"
    >
      Related Videos
    </p>
  </div>
</section>
`;
